package.json

```diff
@@ -28,14 +28,15 @@
     "gulp": "gulp",
     "stage-release": "ts-node --project tools/release/ tools/release/stage-release.ts",
     "publish-release": "ts-node --project tools/release/ tools/release/publish-release.ts",
+    "check-entry-point-setup": "node ./scripts/check-entry-point-setup.js",
```

Comment:
Couldn't we run this with `ts-node` and have it be in TypeScript like we have for the rest of the scripts?

Reply:
I decided to not use TypeScript for the script so that it can be simply run without `ts-node`. I did the same for `build-packages-dist.js` and `deploy-dev-app.js`. I'm not sure what the ideal language is for these scripts. I found everything needed for the scripts in standard JS and it doesn't require the slower `ts-node`.

Reply:
It's not much slower though and most of the other tooling is in TS already. E.g. check the rest of the scripts around it in the `package.json`.

Reply:
The majority of scripts in `scripts/` is JavaScript already. Other tools linked in `package.json` like the release scripts are Typescript-based since they are split up in multiple files and they are not really scripts, but rather `tools/`.

For me personally, the scripts in `./scripts/` should primarily be about usability and not need a preprocessor (unless there is a reason it's required for a particular script). Re slowness: I tried running a simple script and for the simple script I hit quite a difference already.

| ts-node | node |
|-------- | ------- |
| real    **0m1.380s**<br/>user    0m0.030s<br/>sys     0m0.075s | real    **0m0.073s**<br/>user    0m0.000s<br/>sys     0m0.015s | 

I don't mind too much. My point is just that the _majority_ of Node scripts in `./scripts` is already JavaScript based, so we should rather decide on one language for these scripts. This seems out of scope for this PR though?

Reply:
I'm not arguing that it's slower, but it's not something we run manually ourselves and one or two extra seconds is nothing for the build which is taking minutes. I don't mind it either way. We can see what others think about it.

Reply:
The scripts are not necessarily about the build. We could say that the time difference is not impacting much since scripts like `check-rollup-globals` run on CI and the actual JS code is fast.

What's the benefit of using TypeScript for scripts which do not use any TS features? I could imagine consistency (_which is good_). As said above, I don't mind too much either, so if we decide to do it for consistency, then I'm happy to switch all other scripts to TS (it might be good to think about the Bash scripts too)